bluebutton-data-model-rif/src/main/resources/db/migration/V18__Add_inner_join_relationships.sql

```diff
@@ -0,0 +1,105 @@
+/*
+ * Alters the BeneficiariesHistory and MedicareBeneficiaryIdHistory tables to add foreign
+ * keys to Beneficiary.
+ * 
+ * See: https://jira.cms.gov/browse/BLUEBUTTON-865
+ */
```

Comment:
How are we going to test all this to verify that it works as expected?

Reply:
Working with @nw-matt-s to see how we can test this before deploying.

Reply:
Any updates on this?

Reply:
After doing some testing with @nw-matt-s , I've made a few changes and believe we are good to go.

He was able to run the queries for the `BeneficiaryHistory` part and we could see that they were indeed working correctly after a few tweaks. One of the more important changes I'd say is changing the delete on the history tables to only records that exist in the new "invalid" tables. I believe it is more clear, and we know for sure we're not deleting any data we aren't saving elsewhere.

I've saved the tests that we ran in case we need to repeat them, but otherwise I'd say we're looking good.